src/http/client.cr

```diff
@@ -101,18 +105,16 @@ class HTTP::Client
   # ```
   {% if flag?(:without_openssl) %}
     getter! tls : Nil
+    @socket : TCPSocket | Nil
+    alias TLS = Bool | Nil
   {% else %}
     getter! tls : OpenSSL::SSL::Context::Client
+    @socket : TCPSocket | OpenSSL::SSL::Socket | Nil
+    alias TLS = OpenSSL::SSL::Context::Client | Bool | Nil
```

Comment:
I'd prefer to call this `TLSContext` or `TLSConfig`. That's more explicit and avoids issues with a future top-level namespace `TLS`.